packages/smooth_app/lib/pages/user_management/sign_up_page.dart

```diff
@@ -131,9 +132,17 @@ class _SignUpPageState extends State<SignUpPage> {
                 }
                 return null;
               },
+              onSubmit: (String? value) {
+                // Checks if confirm password is equal to password if not then
+                // move the focus to the confirm password field
+                if (_password2Controller.text != value) {
+                  FocusScope.of(context).requestFocus(confirmPassword);
+                }
+              },
             ),
             const SizedBox(height: space),
             SmoothTextFormField(
+              focusNode: confirmPassword,
               type: TextFieldTypes.PASSWORD,
               controller: _password2Controller,
               textInputAction: TextInputAction.next,
```

Comment:
I'm sure your way also works, but have you tried setting the textInputAction for the last field to `TextInputAction.done`, besides of that I can't think of why the focus move doesn't work for the last one

Reply:
So setting the `textInputAction` to `TextInputAction.done` for the confirm password field won't work I guess since the issue starts with the internal mechanism of flutter I guess since the control doesn't get passed automatically ideally the focus passes to the next field inside the for without use of `focusNode` or `textInputAction`. But sometimes this doesn't happen due to some reasons like use of obscure field etc. Then in that case we explicitly need to use this focusNode mechanism.